### ISSUE TITLE
Sponsorblock??

### ISSUE THREAD

**Issue by Yankees4life** [fe80Grau/ytdlp2STRM#4]
Any chance to some sponsorblock implementation in? I have some videos that I could care less about the invid adverts.

Cheers

**Comment by CrazyWolf13** [comment]
Yes Please!

**Comment by fe80Grau** [comment]
hello to both of you again. I am grateful for the participation in my project, I simply do it because I wanted to find something like this on the internet and I never found it the way I wanted, so here I am doing it and sharing my grain of sand. I am really excited that it is used by other people than me

I've messed around a bit with the yt-dlp and sponsorblock options, but it has some problems depending on the version of ffmpeg installed, so I recommend that you replace ffmpeg with the modified version of yt-dlp before trying it, https://github.com /yt-dlp/FFmpeg-Builds#ffmpeg-static-auto-builds . In the README I have written how to do it more or less. I'm working fine with this version and enabling sponsorblock on an Oracle ARM VM running Ubuntu 22.04

In summary, in the last commit I have added two parameters in the config, ytdlp2strm_sponsorblock and ytdlp2strm_sponsorblock_cats.
As the correct functioning of sponsorblock depends on the version of ffmpeg and I'm too lazy to program the check by default, I leave it disabled. To activate it simply change the value of ytdlp2strm_sponsorblock to enable, without quotes and lower case

The ytdlp2strm_sponsorblock_cats parameter is to indicate the segments that you want to eliminate, by default I leave it in sponsor, but in the README I have left the complete list that can be indicated.

I await your comments and that there are not many errors!
